### PR TITLE
fix: `flox search` aliases and deduplication

### PR DESCRIPTION
**Pull request by aakropotkin**
## Proposed Changes

This fixes a bug in `flox search` which caused improper grouping of results under `floxref` headers.

Also fixes a bug that caused column separation to fail for output without `-v`.

The new grouping uses `pname + description` when semver or `-v` is active, and restores the legacy behavior when `-v` is not used.

Drops some `jq` files in favor of a single lib.

Rename `-v,--verbose` to `-l,--long`, but still accept `-v,--verbose` as a "hidden" alias. TODO: deprecate `-v,--verbose`.

## Current Behavior

`floxref` "headers" were improperly set when `-v` was used without semantic version parsing, causing a large number of spurious entries to be listed.

Without `-v` this would fail to deduplicate some results without semantic versioning.

Current arg parser in Rust has a conflict with `flox search -v,--verbose` flag and the "global" option ( `flox -v,--verbose search ...` ) used to control logging.

## Checks

- [x] All tests pass.
- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] I have accepted the flox [Contributor License Agreement](../blob/main/.github/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](../blob/main/.github/CONTRIBUTORS.csv) file by way of this pull request or one done previously.

## Release Notes

`flox search` now groups packages in a more compact manner when `-v,--verbose` is used, and aligns columns when `-v` is not used.

The option `flox search -v,--verbose` has been renamed `-l,--long`. The `-v,--verbose` flag is deprecated and will be removed in the future.

### CODE REVIEW

File: flox-bash/lib/commands/general.sh [hunk 1]
```diff
@@ -224,21 +224,32 @@ function floxSearch() {
 	: "${GREP_COLOR:=1;32}"
 	export GREP_COLOR
 
+	if [[ "$showDetaul" = true ]]; then
```
Comment (tomberek) [inline flox-bash/lib/commands/general.sh:227]:
```suggestion
	if [[ "$showDetail" = true ]]; then
```

File: flox-bash/lib/commands/general.sh [hunk 2]
```diff
@@ -224,21 +224,32 @@ function floxSearch() {
 	: "${GREP_COLOR:=1;32}"
 	export GREP_COLOR
 
+	if [[ "$showDetail" = true ]]; then
+		_m_col="cat -"
+	else
+		_m_col="$_column -s '|' -t -l 2"
```
Comment (garbas) [inline flox-bash/lib/commands/general.sh:230]: I get the following error when running:

```
$ flox build flox
$ ./result/bin/flox search python
Migrating previous telemetry opt out to user config
/nix/store/4vwsj74yv5vzwdzxrvl6rc3gj1bd75q1-util-linux-1003.1-2008/bin/column: illegal option -- l
Usage: column [-tx] [-c columns] [-s sep] [file ...]
warning: input 'flox' has an override for a non-existent input 'nixpkgs-flox'
warning: input 'flox' has an override for a non-existent input 'nixpkgs-flox'
warning: input 'flox' has an override for a non-existent input 'nixpkgs-flox'
```

**Review by garbas — DISMISSED** [review]
This breaks the `flox search` for me. Maybe has something to do with me running on macos.

File: flox-bash/lib/commands/general.sh (hunk 2 above)
Comment (aakropotkin) [reply flox-bash/lib/commands/general.sh:230]: Good catch, this is a difference between the FreeBSD and GNU implementations of `column`.
I'll get it patched.

File: flox-bash/lib/commands/general.sh [hunk 3]
```diff
@@ -181,7 +181,14 @@ function floxSearch() {
 			shift
 			;;
 		# TODO: Deprecate `-v,--verbose'
-		-l|--long|-v|--verbose)
+		-v|--verbose)
+			_msg="the flag 'flox search -v,--verbose' is deprecated."
```
Comment (garbas) [inline flox-bash/lib/commands/general.sh:185]: I think this PR fixes also https://github.com/flox/flox-private/issues/376 so this PR receives bonus points :)

**Review by tomberek — APPROVED** [review]

**Review by garbas — COMMENTED** [review]
The `./result/bin/flox search python` command now works, but I'd add `column` to the hash_commands as we do semver. It feels wrong to rely on system `column` command.

**Pull request merged**